examples/Check group membership of app using JWT and Graph API.policy.xml

```diff
@@ -0,0 +1,107 @@
+
+-->
+<policies>
+    <inbound>
+        <base />
+        <!-- set the group object ID here -->
+        <set-variable name="groupId" value="3e69dd12-469b-4cb7-a361-888eba3ee52c" />
+        <!-- get access token for managed identity -->
+        <authentication-managed-identity resource="https://graph.microsoft.com" output-token-variable-name="msi-access-token" ignore-error="false" />
+        <!-- get tenant id -->
+        <set-variable name="tenantId" value="@{
+            Jwt jwt;
+            ExpressionSyntaxExtensions.TryParseJwt((string)context.Variables["msi-access-token"], out jwt);
+            return jwt.Claims.GetValueOrDefault("tid", "");
+        }" />
+        <!-- validate the incoming JWT token -->
+        <validate-jwt header-name="Authorization" output-token-variable-name="valid-jwt">
+            <openid-config url="@("https://login.microsoftonline.com/" + context.Variables["tenantId"] + "/.well-known/openid-configuration")" />
+            <audiences>
+                <audience>api://NpTestBackendAPI/</audience>
+            </audiences>
+            <issuers>
+                <issuer>@("https://sts.windows.net/" + context.Variables["tenantId"] + "/")</issuer>
+            </issuers>
+        </validate-jwt>
+        <!-- get app id from validated JWT to use as the caching key -->
+        <set-variable name="appId" value="@{
+            var jwt = (Jwt)context.Variables["valid-jwt"];
+            return jwt.Claims.GetValueOrDefault("appid", "");
+        }" />
+        <!-- Check in cache if this app id has already been checked, otherwise check and then add to cache -->
+        <cache-lookup-value key="@((string)context.Variables["appId"])" variable-name="membershipCached" />
+        <choose>
+            <when condition="@(!context.Variables.ContainsKey("membershipCached"))">
+                <!-- check for membership of the app in a particular AAD group, using the app's object ID from the validated JWT -->
+                <send-request ignore-error="false" timeout="20" response-variable-name="inGroupResult" mode="new">
+                    <set-url>@{
+                        var jwt = (Jwt)context.Variables["valid-jwt"];
+                        return "https://graph.microsoft.com/v1.0/directoryObjects/" + jwt.Claims.GetValueOrDefault("oid", "") + "/checkMemberGroups";
+                    }</set-url>
+                    <set-method>POST</set-method>
+                    <set-header name="Content-Type" exists-action="override">
+                        <value>application/json</value>
+                    </set-header>
+                    <set-header name="Authorization" exists-action="override">
+                        <value>@("Bearer " + (string)context.Variables["msi-access-token"])</value>
+                    </set-header>
+                    <set-body>@("{'groupIds': ['" + context.Variables["groupId"] + "']}")</set-body>
```

Comment:
Should use JObject here because if groupId needs quoting, it won't be done and will result in bad JSON.